component/src/main/java/com/synopsys/integration/alert/component/settings/proxy/validator/SettingsProxyValidator.java

```diff
@@ -0,0 +1,56 @@
+package com.synopsys.integration.alert.component.settings.proxy.validator;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.descriptor.config.field.errors.AlertFieldStatus;
+import com.synopsys.integration.alert.common.descriptor.config.field.errors.AlertFieldStatusMessages;
+import com.synopsys.integration.alert.common.rest.model.ValidationResponseModel;
+import com.synopsys.integration.alert.component.settings.proxy.model.SettingsProxyModel;
+
+@Component
+public class SettingsProxyValidator {
+    public static final String PROXY_HOST_FIELD_NAME = "proxyHost";
+    public static final String PROXY_PORT_FIELD_NAME = "proxyPort";
+    public static final String PROXY_USERNAME_FIELD_NAME = "proxyUsername";
+    public static final String PROXY_PASSWORD_FIELD_NAME = "proxyPassword";
+
+    public ValidationResponseModel validate(SettingsProxyModel model) {
+        Set<AlertFieldStatus> statuses = new HashSet<>();
+        if (model.getProxyHost().isPresent() && model.getProxyPort().isEmpty()) {
+            validateRequiredFieldIsNotBlank(statuses, model.getProxyPort().isPresent(), PROXY_PORT_FIELD_NAME);
+        }
```

Comment:
We may want to introduce a new class to help with common validation similar to what ConfigurationFieldValidator does today otherwise we'd end up with the same private method in many places. The new class could then be used in EmailGlobalConfigurationValidator.

Reply:
The new class could also return the content in the form of a validation response so you don't need to do the statuses.empty check again.

Reply:
While this doesn't seem like a bad idea, I think it might be out of the scope of this PR. I can create a ticket to look at the existing implementations for Email, Encryption, and Proxy to determine what would be the best way to write a common validator. Right now each validator is somewhat different and it may take additional time to investigate, come up with a solution and write new tests around all of them.